core/ustring.cpp

```diff
@@ -3755,8 +3755,8 @@ String String::get_file() const {
 String String::get_extension() const {
 
 	int pos = find_last(".");
-	if (pos < 0)
-		return *this;
+	if (pos < 0 || pos < MAX(find_last("/"), find_last("\\")))
```

Comment:
Just for the record, in my implementation I had it just as `if (pos <= MAX(...))`.

Reply:
I don't think it'll make a huge difference.  
Let me know if you want me to change this before it's merged 😉

Reply:
No, don't change it, the two are equivalent for all the possible results of `pos` and `MAX`.

On the other hand, something you might change (if you really want to change something) is to make the function return `String()` instead of `""`.